app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/ParseLinkConnectionDetailsUseCase.kt

```diff
@@ -12,60 +11,79 @@ import com.radixdlt.sargon.PublicKey
 import com.radixdlt.sargon.RadixConnectPassword
 import com.radixdlt.sargon.extensions.hexToBagOfBytes
 import com.radixdlt.sargon.extensions.init
+import kotlinx.serialization.json.Json
 import rdx.works.core.decodeHex
 import rdx.works.profile.data.model.apppreferences.P2PLinkPurpose
 import rdx.works.profile.ret.crypto.PrivateKey
 import timber.log.Timber
 import javax.inject.Inject
 
 class ParseLinkConnectionDetailsUseCase @Inject constructor(
-    private val p2PLinksRepository: P2PLinksRepository
+    private val p2PLinksRepository: P2PLinksRepository,
+    private val json: Json
```

Comment:
not that it's something wrong with this, but why you pass json in constructor of your usecases?

Reply:
There is a comment from @jakub-rdx suggesting this and I agree with it for the sake of reducing static calls, even though we don't benefit much from it here.

But I guess depending on the serializer class rather than calling static functions, can be seen as the first step towards something more robust in the future, like: instead of having Json reference here directly, we'd have an interface type injected here (agnostic of any serialization library), which would eventually rely upon Json so that we could easily mock the output of encoding/decoding functions or easily replace the serialization library if we'd ever have to.